snuba/reader.py

```diff
@@ -31,6 +28,36 @@
 else:
     Result = MutableMapping[str, Any]
 
+
+def iterate_rows(result: Result) -> Iterator[Row]:
+    if "totals" in result:
+        return itertools.chain(result["data"], [result["totals"]])
+    else:
+        return iter(result["data"])
+
+
+def build_result_transformer(
+    column_transformations: Mapping[str, Callable[[Any], Any]],
+) -> Callable[[Result], None]:
+    """
+    Builds and returns a function that can be used to mutate a ``Result``
+    instance in-place by transforming all values for columns that have a
+    transformation function specified for their data type.
+    """
+
+    def transform_result(result: Result) -> None:
+        for column in result["meta"]:
+            transformer = column_transformations.get(column["type"])
```

Comment:
Note that this still doesn't handle tuples (not a regression), e.g. if `argMax((timestamp, value))` was provided in the aggregations, the query would succeed but it would fail to be serialized (either into the cache, or into the HTTP response buffer.)